internal/crop/crop.go

```diff
@@ -0,0 +1,62 @@
+package crop
+
+import (
+	"image"
+	"image/draw"
+)
+
+// An interface that is
+// image.Image + SubImage method.
+type subImageSupported interface {
+	SubImage(r image.Rectangle) image.Image
+}
+
+// Cropper is the interface used to crop images
+type Cropper interface {
+	Crop(img image.Image, width, height int, anchor image.Point) (image.Image, error)
+}
+
+// CropperFunc exposes a Crop function that calls itself.
+// It implements Cropper.
+type CropperFunc func(img image.Image, width, height int, anchor image.Point) (image.Image, error)
+
+// Crop calls the CropperFunc
+func (c CropperFunc) Crop(img image.Image, width, height int, anchor image.Point) (image.Image, error) {
+	return c(img, width, height, anchor)
+}
+
+// Crop retrieves an image that is a
+// cropped copy of the original img.
+func Crop(img image.Image, width, height int, anchor image.Point) (image.Image, error) {
+	maxBounds := maxBounds(anchor, img.Bounds())
+	size := computeSize(maxBounds, image.Point{width, height})
```

Comment:
From the logic, there is no need to create maxBounds / computeSize functions.
just need to use 
``` size := image.Point{width, height}```